client/cody/src/completions/providers/anthropic.ts

```diff
@@ -155,38 +95,93 @@ export class AnthropicProvider extends Provider {
             remainingChars -= numSnippetChars
         }
 
-        return [...referenceSnippetMessages, ...prefixMessages]
+        return { messages: [...referenceSnippetMessages, ...prefixMessages], prefix }
     }
 
-    public async generateCompletions(abortSignal: AbortSignal): Promise<Completion[]> {
-        // TODO: Bring back the logic with injectPrefix \n when the current line is non empty
+    private postProcess(rawResponse: string, prefix: PrefixComponents): string {
+        let completion = extractFromCodeBlock(rawResponse)
+
+        if (prefix.tail.rearSpace.length > 0) {
+            const rearSpaceLines = prefix.tail.rearSpace.split('\n')
+            const currentLine = rearSpaceLines[rearSpaceLines.length - 1]
+            if (currentLine.match(/^\s*$/)) {
+                // New line + indent: trim all leading new lines and match indent
+                completion = completion.replace(/^\n*/, '')
+                const completionIndent = completion.match(/^\s*/)![0]
+                if (currentLine.length <= completionIndent.length) {
+                    completion = completionIndent.slice(currentLine.length) + completion.slice(completionIndent.length)
+                }
+            }
+        }
+
+        completion = trimIndent(completion, this.suffix)
+        completion = trimUntilSuffix(completion, this.suffix)
+
+        // Remove incomplete lines in single-line completions
+        if (this.multilineMode !== 'block') {
+            const allowedNewlines = 2
+            const lines = completion.split('\n')
+            if (lines.length >= allowedNewlines) {
+                completion = lines.slice(0, allowedNewlines).join('\n')
+            }
+        }
+
+        completion = completion.trimEnd()
+        return completion
+    }
 
+    public async generateCompletions(abortSignal: AbortSignal, n?: number): Promise<Completion[]> {
```

Comment:
```suggestion
    public async generateCompletions(abortSignal: AbortSignal): Promise<Completion[]> {
```

That's a leftover from the rebase I think, `n` is now a private member of the class